- Disable breaking if no haptic feedback support which disallows gamecontroller API
- Allow custom SDL GetPlatform name with env var SDL_PLATFORM

Index: src/SDL.c
--- src/SDL.c.orig
+++ src/SDL.c
@@ -301,9 +301,6 @@ SDL_InitSubSystem(Uint32 flags)
         }
         SDL_PrivateSubsystemRefCountIncr(SDL_INIT_GAMECONTROLLER);
         flags_initialized |= SDL_INIT_GAMECONTROLLER;
-#else
-        SDL_SetError("SDL not built with joystick support");
-        goto quit_and_error;
 #endif
     }
 
@@ -533,6 +530,10 @@ SDL_GetRevisionNumber(void)
 const char *
 SDL_GetPlatform(void)
 {
+    const char *platform;
+    if (platform = SDL_getenv("SDL_PLATFORM")) {
+        return platform;
+    }
 #if __AIX__
     return "AIX";
 #elif __ANDROID__
